### PR TITLE
[#10172] improvement(server): Guard catch-block request dereferences in REST handlers

### LINKED ISSUE #10172: [Improvement] Catch-path request dereference can mask original exceptions in REST handlers
### What would you like to be improved?

Multiple REST handlers dereference `request` inside `catch (Exception e)` (for example `request.getName()`, `request.getRoleNames()`, `request.getUpdates()`, `request.getDrops()`, `request.getJobTemplate().name()`). When request is `null`, catch-path code can throw a second `NullPointerException`, masking the real failure.

Representative failure:
- Primary exception occurs in method body.
- Catch block dereferences `request.get*()`.
- Secondary `NullPointerException` in catch hides the original exception context.

## Scope
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/TableOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/FilesetOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/FunctionOperations.java` register catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/ModelOperations.java` register catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/SchemaOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/MetalakeOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/CatalogOperations.java` create/testConnection catch paths
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/UserOperations.java` add catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/GroupOperations.java` add catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/RoleOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/PolicyOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/TagOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/TopicOperations.java` create catch path
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/PermissionOperations.java` grant/revoke catch paths
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/StatisticOperations.java` update/drop (+partition update/drop) catch paths
- [ ] `server/src/main/java/org/apache/gravitino/server/web/rest/JobOperations.java` register-job-template catch path

## Expected behavior
- No `request.get*()` dereference inside affected `catch` blocks.
- Catch blocks should always preserve/marshal the primary error through existing `ExceptionHandlers`.
- Endpoint APIs and response contracts remain unchanged.

## Definition of done
- [ ] Replace catch-path request dereferences with null-safe precomputed identifiers or safe fallbacks.
- [ ] Keep existing `ExceptionHandlers` mapping behavior.
- [ ] Keep endpoint APIs unchanged.
- [ ] Add/adjust tests to ensure exception handling path is stable when request is null.
- [ ] Run and pass: `./gradlew test -PskipITs`

## Example unit test
```java
@Test
public void testAddGroupWithNullRequestBodyDoesNotExposeNpe() {
  Response resp =
      target("/metalakes/metalake1/groups")
          .request(MediaType.APPLICATION_JSON_TYPE)
          .accept("application/vnd.gravitino.v1+json")
          .post(Entity.entity("null", MediaType.APPLICATION_JSON_TYPE));

  Assertions.assertEquals(Status.INTERNAL_SERVER_ERROR.getStatusCode(), resp.getStatus());
  ErrorResponse error = resp.readEntity(ErrorResponse.class);
  Assertions.assertNotEquals(NullPointerException.class.getSimpleName(), error.getType());
}
```

### PR DESCRIPTION
**Pull request by sachinnn99**
### What changes were proposed in this pull request?

Multiple REST handlers dereference `request` (e.g. `request.getName()`, `request.getRoleNames()`, `request.getJobTemplate().name()`) inside `catch (Exception e)` blocks. When `request` is `null` (e.g. from an empty or malformed POST body), the catch block throws a secondary `NullPointerException` that masks the real failure.

This PR replaces all catch-path request dereferences with null-safe precomputed local variables across 16 REST handler files:

- Simple `getName()` pattern (11 files): Table, Fileset, Function, Model, Schema, Catalog, User, Group, Role, Policy, Tag, Topic
- `getRoleNames()` pattern: PermissionOperations (4 catch sites)
- Nested `getJobTemplate().name()`: JobOperations
- `isInUse()` / `isEnable()` pattern: MetalakeOperations, CatalogOperations, PolicyOperations
- `getNames()` pattern: StatisticOperations

**Note on StatisticOperations scope:** The issue lists 4 methods (`updateStatistics`, `dropStatistics`, `updatePartitionStatistics`, `dropPartitionStatistics`), but only `dropStatistics` is actually vulnerable — the other 3 already use null-safe helpers (`getStatisticNames`, `getPartitionNames`, `getDropPartitionNames`).

### Why are the changes needed?

When `request` is null and an exception occurs in the method body, the catch block tries to dereference `request.get*()`, throwing a secondary NPE that masks the original exception. This makes debugging harder as the real error is lost. The fix ensures catch blocks always preserve the primary error through existing `ExceptionHandlers`.

Additionally, `JobOperations.registerJobTemplate` and `PolicyOperations.setPolicy` had pre-try `request` dereferences (outside the try-catch entirely), meaning a null request would NPE before the catch block could handle it at all.

Reproducible by sending an empty body to any affected endpoint:
```
POST /api/metalakes/m/catalogs/c/schemas
Content-Type: application/json
Accept: application/vnd.gravitino.v1+json
(empty body)
```

Fix: #10172

### Does this PR introduce _any_ user-facing change?

No. Error handling behavior is preserved — endpoints return the same error responses, but the original exception is no longer masked by a secondary NPE.

### How was this patch tested?

- Added null-request-body regression tests covering all main dereference patterns:
  - `TestGroupOperations.testAddGroupWithNullRequest` — `getName()` pattern
  - `TestPermissionOperations.testGrantRolesToUserWithNullRequest` — `getRoleNames()` pattern
  - `TestCatalogOperations.testCreateCatalogWithNullRequest` — `getName()` pattern in CatalogOperations
  - `TestJobOperations.testRegisterJobTemplateWithNullRequest` — nested `getJobTemplate().name()` pattern
- All existing tests pass: `./gradlew :server:test -PskipITs`

### CODE REVIEW

**Comment by sachinnn99** [comment]
This builds on the same approach as #10183. Changes beyond that PR:

- Guards nested fields (e.g. `getJobTemplate().name()` in `JobOperations`)
- Fixes pre-try dereferences in `JobOperations.registerJobTemplate` and `PolicyOperations.setPolicy` that would NPE before the catch block
- Adds regression tests for the `getName()`, `getRoleNames()`, and `getJobTemplate().name()` patterns

File: server/src/test/java/org/apache/gravitino/server/web/rest/TestCatalogOperations.java [hunk 1]
```diff
@@ -588,6 +588,22 @@ public void testSetCatalog() {
     Assertions.assertEquals(RuntimeException.class.getSimpleName(), errorResponse1.getType());
   }
 
+  @Test
+  public void testCreateCatalogWithNullRequest() {
+    Response resp =
+        target("/metalakes/metalake1/catalogs")
+            .request(MediaType.APPLICATION_JSON_TYPE)
+            .accept("application/vnd.gravitino.v1+json")
+            .post(Entity.entity("null", MediaType.APPLICATION_JSON_TYPE));
```
Comment (jerryshao) [inline server/src/test/java/org/apache/gravitino/server/web/rest/TestCatalogOperations.java:597]: As I mentioned in other PR, parsing "null" will not normally happen, I really doubt the necessity of such a fix.

Another thing, if we really want to fix this, it is better to fix this in the object mapper. We should probably not allow such deserialization.

Comment (sachinnn99) [reply server/src/test/java/org/apache/gravitino/server/web/rest/TestCatalogOperations.java:597]: Thanks @jerryshao. I looked into a centralized fix as you suggested:

- Jackson has no `DeserializationFeature` for rejecting null top-level POJO deserialization
- `ContainerRequestFilter` runs before deserialization, so it can't catch `"null"` JSON (the body is present, it just deserializes to Java `null`)
- The viable centralized options are Bean Validation (`@NotNull` + `jersey-bean-validation`) or a custom `MessageBodyReader` — both are worth exploring but are larger scope than this issue

I'll close this PR for now. If you'd be open to a centralized approach, I can open a separate issue for that.

Comment (jerryshao) [reply server/src/test/java/org/apache/gravitino/server/web/rest/TestCatalogOperations.java:597]: Yes, I would prefer a centralized solution.
